apiv2/db_import/tests/populate_db.py

```diff
@@ -69,227 +67,278 @@ def stale_deposition_author() -> dict:
         "primary_author_status": False,
     }
 
+def write_data(function):
+    def wrapper(session: sa.orm.Session, *args, **kwargs):
+        obj = function(session, *args, **kwargs)
+        if obj:
+            session.add(obj)
+        session.flush()
+    return wrapper
+
 
-def populate_deposition_authors() -> None:
-    DepositionAuthor.create(**stale_deposition_author())
+@write_data
+def populate_deposition(session: sa.orm.Session) -> Deposition:
```

Comment:
1. I'm assuming the order you call these `populate` functions matters since it's building relationships within the database. If the relationships are not setup correctly I image the commit with fail right? That requirement is not clear from reading the code alone. We could either point the user to the documentation of the database schema or code the ordering in additional populate functions. I prefer coding it.

2. It would be easier to manage if we made the `populate` functions pytest fixtures. The only parameter they use is `session` which is already a pytest fixture, and the data is hardcoded.

3.  We can also simplify some of the tests a little implement both 1 and 2 by commiting the session in the fixture. Then we could remove `sync_db_session` from most of the tests.

Reply:
will be resolved in https://github.com/chanzuckerberg/cryoet-portal/issues/121